Microsoft.Toolkit.HighPerformance/Extensions/BoolExtensions.cs

```diff
@@ -21,9 +21,9 @@ public static class BoolExtensions
         /// <remarks>This method does not contain branching instructions.</remarks>
         [Pure]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static byte ToByte(this bool flag)
+        public static unsafe byte ToByte(this bool flag)
         {
-            return Unsafe.As<bool, byte>(ref flag);
+            return *(byte*)&flag;
```

Comment:
New optimization from .NET 5 compiler???

Reply:
Ahahah nope, just good old C# trickery 😄
This was actually a suggestion from Tanner, the idea here is that the two versions do exactly the same anyway, but using raw pointers means the GC doesn't have to also track the additional metadata to ensure that if a GC collection happens while this method is running, the locals are properly handled. In our case that's not needed since the target values are already pinned (they're just local arguments passed by copy). So using raw pointers will produce the same exact codegen, but with some theoretical minor indirect benefits on top of that 👍